lib/ops/opsservice/operationgroup.go

```diff
@@ -121,6 +121,13 @@ func (g *operationGroup) canCreateOperation(operation ops.SiteOperation) error {
 		default:
 			return trace.CompareFailed("the cluster is %v", cluster.State)
 		}
+	case ops.OperationGarbageCollect:
```

Comment:
Nit: you could've just added it to the previous case clause since the conditions seem to be the same:

```go
case ops.OperationShrink, ops.OperationGarbageCollect:
    // shrink and gc are allowed for degraded clusters
```